From: SJ Park <sj@kernel.org>
Date: Sat, 8 Aug 2026 10:02:46 -0700
Subject: [PATCH] ==== refactor damon_commit for zero quota target_value ====

mm/damon: move zero damos quota target_value handling to the core layer

Having zero DAMOS quota target value can cause division by zero.  DAMON
API callers are checking the target value parameters to avoid that.  It
is easy to make mistakes in some of the multiple API callers.  Move that
to the core layer.

Patch 1 adds the corner case handling into the core layer DAMON
parameters validation logic.  Patches 2 and 3 remove no more needed
DAMON API callers side handling of the corner case in DAMON_LRU_SORT and
DMON_SAMPLIE_MTIER, respectively.

Signed-off-by: SJ Park <sj@kernel.org>
---
 hkml_cv_bogus/hkml_cv_bogus_hif56cht | 0
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 hkml_cv_bogus/hkml_cv_bogus_hif56cht

diff --git a/hkml_cv_bogus/hkml_cv_bogus_hif56cht b/hkml_cv_bogus/hkml_cv_bogus_hif56cht
new file mode 100644
index 0000000000000..e69de29bb2d1d
-- 
2.47.3

